src/shared/gravity_value.c

```diff
@@ -2192,6 +2192,99 @@ void gravity_map_blacken (gravity_vm *vm, gravity_map_t *map) {
+//     if (VALUE_ISA_STRING(value))
+// }
+
+static void gravity_map_stringify_iterator(gravity_hash_t *hashtable, gravity_value_t key, gravity_value_t v, void *data) {
+    #pragma unused(hashtable)
+    // TODO: make extra json characters length a constant, right now it's 10
+
+    // keys are always strings
+    assert(key.isa == gravity_class_string);
+    gravity_string_t *map_string = (gravity_string_t *)data;
+    char *substr;
+    char *key_string = VALUE_AS_CSTRING(key);
+    char *delimiter = ", ";
+
+    // if we just started, open with a curly brace
+
+    if (!strlen(map_string->s)) {
+        gravity_string_concat_cstring(NULL, map_string, "{");
+        delimiter = "";
+    }
+
+    // char *temp_map_string = mem_alloc(NULL, strlen(*map_string) * sizeof(char));
+    // BOOL
+    if (VALUE_ISA_BOOL(v)) {
+        substr = mem_alloc(NULL, strlen(key_string) + 11 * sizeof(char));
+        sprintf(substr, "%s\"%s\": %s", delimiter, key_string, (v.n == 0) ? "false" : "true");
+        gravity_string_concat_cstring(NULL, map_string, substr);
+        return;
+    }
+
+    // INT
+    if (VALUE_ISA_INT(v)) {
+        substr = mem_alloc(NULL, strlen(key_string) + sizeof(int64_t) + 10 * sizeof(char));
+        sprintf(substr, "%s\"%s\": %lld", delimiter, key_string, (int64_t)v.n);
+        gravity_string_concat_cstring(NULL, map_string, substr);
+        return;
+    }
+
+    // // FLOAT
+    if (VALUE_ISA_FLOAT(v)) {
+        substr = mem_alloc(NULL, strlen(key_string) + sizeof(float) + 10 * sizeof(char));
+        sprintf(substr, "%s\"%s\": %f", delimiter, key_string, v.f);
+        gravity_string_concat_cstring(NULL, map_string, substr);
+        return;
+    }
+
+    // STRING
+    if (VALUE_ISA_STRING(v)) {
+        char *value_string = VALUE_AS_CSTRING(v);
+        substr = mem_alloc(NULL, strlen(key_string) + strlen(value_string) + 10 * sizeof(char));
+        sprintf(substr, "%s\"%s\": \"%s\"", delimiter, key_string, value_string);
+        gravity_string_concat_cstring(NULL, map_string, substr);
+        return;
+    }
+
+    // MAP
+    if (VALUE_ISA_MAP(v)) {
+        gravity_map_t *value_map = VALUE_AS_MAP(v);
+        char *value_map_string = gravity_map_to_string(NULL, value_map);
+        substr = mem_alloc(NULL, strlen(key_string) + strlen(value_map_string) + 10 * sizeof(char));
+        sprintf(substr, "%s\"%s\": %s", delimiter, key_string, value_map_string);
+        gravity_string_concat_cstring(NULL, map_string, substr);
+        return;
+    }
+
+    // should never reach this point
+    assert(0);
+}
+
+char *gravity_map_to_string(gravity_vm *vm, gravity_map_t *map) {
+    gravity_string_t *map_string = gravity_string_new(NULL, "", 1000, 0);
+    gravity_hash_iterate(map->hash, gravity_map_stringify_iterator, map_string);
+    gravity_string_concat_cstring(NULL, map_string, "}");
+    return map_string->s;
+}
+
+gravity_value_t gravity_map_to_value (gravity_vm *vm, gravity_map_t *map) {
```

Comment:
Not sure to understand the use case of this...

Reply:
@marcobambini Wouldn't you say there's a need for _to_value functions for all types? E.g. if I want to create a map with an key associated to an array, I'd have no way of doing that without doing something like:

```c
gravity_map_t *response = gravity_map_new(vm, 32);
gravity_list_t *headers = gravity_list_new(vm, 32);
gravity_map_insert(vm, response,
    gravity_string_to_value("Headers"),
    gravity_list_to_value(headers));
```

If there's a way for already doing something like this, let me know. I'm cleaning up the implementation at the moment, currently trying to get something like this going

Reply:
I use the map_to_value method in a similar fashion here: https://github.com/marcobambini/gravity/pull/281/files#diff-fb63d91a9c6432f884cfba6cc100f7d9R160

Reply:
Oh I guess VALUE_FROM_OBJECT might be helpful in this case 😎